An attempt at in-memory undo/redo support

aaronSig opened the pull request:
This add a hook & storybook example showing in-memory undo and redo support. It works by wrapping the `getCellContent` and `setCellValue` functions so it can fetch a cell before it is updated and replay any updates back. 

Open to feedback and no pressure to merge if you don't think this is in scope. It may be better done in the datasource.

jassmith commented on packages/core/src/data-editor/stories/data-editor-beautiful.stories.tsx:
```diff
@@ -3612,3 +3613,54 @@ export const DropEvents: React.VFC = () => {
+        </BeautifulWrapper>
+    );
+};
+(UnevenRows as any).parameters = {
```
Wrong story :)

aaronSig replied:
Ah - thanks for catching.

jassmith reviewed (changes requested):
You are right about moving this to the source package. Other than that this looks fine to me. We can work on improvements incrementally.

aaronSig commented:
That's great! 

I've fixed up that typo & moved the hook down into the source package. I needed some of the utils for the story so I've moved those as well.

tortarino commented:
Hello,

Thanks to both of you for all the great work!

It seems the undo/redo feature has not made it to the npm repository:
https://www.npmjs.com/package/@glideapps/glide-data-grid-source?activeTab=code

Am I missing something?

Thanks

aaronSig commented:
No, it doesn't look like it's in npm. 

The undo/redo hook is all self contained in one file though `packages/source/src/use-undo-redo.ts`

You could get by just copying that into your project.
